### PR TITLE
Element links

### PR DESCRIPTION
**Pull request by accuminium**
Last but not least in the current doc usability sprint, I propose adding links to parameters, ports, and signals.

### CODE REVIEW

File: src/doc/html/mod.rs [hunk 1]
```diff
@@ -330,3 +334,32 @@ impl<'a> Renderer<'a> {
...
+
+impl Id for TypeItem {
+    fn html_id(&self) -> String {
+        format!("type.{}", self.name)
+    }
+}
+
+impl Id for VarItem {
+    fn html_id(&self) -> String {
+        format!("signal.{}", self.name)
```
Comment (accuminium) [inline src/doc/html/mod.rs:363]: We are currently using `signal` as prefix in the ID of all `VarItems`. I wonder if we should change that to `var` to be closer to the language.

File: src/doc/html/mod.rs [hunk 2]
```diff
@@ -229,9 +229,10 @@ impl<'a> Renderer<'a> {
             for i in &cx.params {
                 write!(
                     out,
-                    "<h3 id=\"parameter.{0}\" class=\"impl\"><code class=\"in-band\">{0}<span class=\"type-annotation\">: {1}</span></code></h3>",
+                    "<h3 id=\"{2}\" class=\"impl\"><code class=\"in-band\"><a href=\"#{2}\">{0}</a><span class=\"type-annotation\">: {1}</span></code></h3>",
```
Comment (accuminium) [inline src/doc/html/mod.rs:232]: Not directly related, but all those `<h3 ...` strings are very similar and their construction should be refactored IMO.

Comment (fabianschuiki) [reply src/doc/html/mod.rs:232]: Good point...

File: src/doc/html/mod.rs (hunk 1 above)
Comment (fabianschuiki) [reply src/doc/html/mod.rs:363]: Yeah we could do that!

**Comment by zarubaf** [comment]
Thanks, LGTM. I've pushed two new issues on the topics raised by you. Going to merge this now and we can address the remaining issues in separate PRs.

**Pull request merged**